manifests/params.pp

```diff
@@ -31,4 +31,26 @@
   $proxy_port     = undef
   $proxy_username = undef
   $proxy_password = undef
+
+  $num_workers = min($::processorcount, 8)
+
+  case $::operatingsystem {
+    'RedHat': {
+      case $::operatingsystemmajrelease {
```

Comment:
parsing operatingsystemrelease appears to be the most common method. We certainly use it all over the place in theforeman modules.